storage/src/main/java/com/google/cloud/trace/zipkin/StackdriverSpanConsumer.java

```diff
@@ -33,15 +43,81 @@
 
   private final TraceTranslator translator;
   private final TraceConsumer consumer;
+  private final AsyncListenableTaskExecutor executor;
 
-  public StackdriverSpanConsumer(TraceTranslator translator, TraceConsumer consumer) {
+  public StackdriverSpanConsumer(
+      TraceTranslator translator, TraceConsumer consumer, AsyncListenableTaskExecutor executor) {
     this.translator = translator;
     this.consumer = consumer;
+    this.executor = executor;
   }
 
   @Override
-  public void accept(List<Span> spans) {
-    Collection<Trace> traces = translator.translateSpans(spans);
-    consumer.receive(Traces.newBuilder().addAllTraces(traces).build());
+  public Call<Void> accept(List<Span> spans) {
+    final Traces traces = Traces.newBuilder().addAllTraces(translator.translateSpans(spans)).build();
+    try {
+      ListenableFuture<Void> future =
```

Comment:
it might be slightly more direct to not use ListenableFuture here as for example call.execute can just call `consumer.receive(traces);` directly.  I'd probably do it that way, extending from base call (which we ought to make a public type :P )

in any case, the point of returning call is to return something not started, so I'd move it if you can.

Reply:
Got it - a little annoying duplicating all the state from the `Future` but went ahead and moved the execution into the `Call`, PTAL